### PR TITLE
Feature/logging

### PR DESCRIPTION
**Pull request by ludomori99**
Add few lines in base.py to somewhat store the prompts. Large room for improvement in details of logging, however not straightforward due to convoluted structure of chaining. it's good to have something working for 801 deployment.

### CODE REVIEW

File: A2rchi/chains/base.py [hunk 1]
```diff
@@ -88,19 +91,27 @@ def from_llm(
         combine_docs_chain_kwargs = combine_docs_chain_kwargs or {}
         _prompt = QA_PROMPT
         document_variable_name = "context"
+
+        logfile = data_path+"chain_input_output.log"
```
Comment (julius-heitkoetter) [inline A2rchi/chains/base.py:95]: This should probably be done using `os.path.join()`. This way it doesn't matter if `data_path` has a "/" on the end of it or not.

File: A2rchi/chains/base.py [hunk 2]
```diff
@@ -88,19 +91,27 @@ def from_llm(
...
+
+        logger.add(logfile, colorize=True, enqueue=True)
```
Comment (julius-heitkoetter) [inline A2rchi/chains/base.py:97]: For documentation purposes, could you add a comment or two above this explaining what this logging is doing specifically.

File: A2rchi/chains/base.py (hunk 1 above)
Comment (julius-heitkoetter) [reply A2rchi/chains/base.py:95]: Also, may want to consider "chain_input_output.log" to be a configurable parameter. But I'll leave this to you.

For context, I've been meaning to change "conversations.json" to a configurable parameter.

If you do put it into the config, make sure to write a comment on what it is and any specifications (if it has to end in .log, etc.)

Not required though, just a thought

**Review by julius-heitkoetter — CHANGES_REQUESTED** [review]
Looks good! 2 things that I believe are needed and then one more thing which you may want (or not want) to consider.

File: A2rchi/chains/base.py (hunk 2 above)
Comment (mdr223) [reply A2rchi/chains/base.py:98]: Looks good now

**Review by mdr223 — APPROVED** [review]
I believe this is ready to be merged 👍

**Pull request merged**